### ISSUE TITLE
Compilation error - commit 5de3447 and newer

### ISSUE THREAD

**Issue by sirlucjan** [systemd/systemd#3035]
### Submission type
- [x] Bug report

_NOTE: Do not submit anything other than bug reports or RFEs via the issue tracker!_
### systemd version the issue has been seen with

> 229 - git version -  commit 5de3447 and newer

_NOTE: Do not submit bug reports about anything but the two most recently released systemd versions upstream!_
### Used distribution

> Arch Linux (kernel 4.4.5, libxkbcommon 0.6.1)
### In case of bug report: Expected behaviour you didn't see

> Commits 5de3447 and newer don't build properly. Commits 198402d and older built and worked fine. 

```
[lucjan@archlinux ~]$ pacman -Ss systemd
lucjan-experimental/libsystemd-git 229.r626.g198402d-1
    systemd client libraries
lucjan-experimental/systemd-git 229.r626.g198402d-1
    system and service manager
lucjan-experimental/systemd-sysvcompat-git 229.r626.g198402d-1 (base)
    sysvinit compat for systemd
```
### In case of bug report: Unexpected behaviour you saw

```
In function '__ppoll_alias',
    inlined from 'bus_poll' at src/libsystemd/sd-bus/sd-bus.c:2871:11:
/usr/include/bits/poll2.h:71:9: warning: call to '__ppoll_chk_warn' declared with attribute warning: ppoll called with fds buffer too small file nfds entries
  return __ppoll_chk (__fds, __nfds, __timeout, __ss, __bos (__fds));
         ^
/tmp/ccJeqdXK.ltrans18.ltrans.o:<artificial>:function method_set_x11_keyboard: error: undefined reference to 'dlopen'
/tmp/ccJeqdXK.ltrans18.ltrans.o:<artificial>:function method_set_x11_keyboard: error: undefined reference to 'dlvsym'
/tmp/ccJeqdXK.ltrans18.ltrans.o:<artificial>:function method_set_x11_keyboard: error: undefined reference to 'dlvsym'
/tmp/ccJeqdXK.ltrans18.ltrans.o:<artificial>:function method_set_x11_keyboard: error: undefined reference to 'dlvsym'
/tmp/ccJeqdXK.ltrans18.ltrans.o:<artificial>:function method_set_x11_keyboard: error: undefined reference to 'dlvsym'
/tmp/ccJeqdXK.ltrans18.ltrans.o:<artificial>:function method_set_x11_keyboard: error: undefined reference to 'dlclose'
/tmp/ccJeqdXK.ltrans18.ltrans.o:<artificial>:function method_set_x11_keyboard: error: undefined reference to 'dlclose'
collect2: error: ld returned 1 exit status
```
### In case of bug report: Steps to reproduce the problem
1. Download commit 5de3447 or newer
2. Command "make"
3. Got an error

**Comment by evverx** [comment]
Try `-ldl` :)

**Comment by sirlucjan** [comment]
Works fine.

**Comment by poettering** [comment]
Fixed by #3036.
